File: back-end/src/schemas/inventory.py
from pydantic import BaseModel


class AddInventoryRequest(BaseModel):
    """Body for POST /api/inventory/"""

    name: str
    quantity: int


class AddInventoryResponse(BaseModel):
    """Response for POST /api/inventory/ (202)."""

    id: int
    identifier: str


class InventoryItem(BaseModel):
    """Single inventory item (GET /api/inventory/<id>/)."""

    id: int
    identifier: str
    name: str | None = None
    quantity: int | None = None
    last_updated: str | None = None

File: front-end/src/types/inventory.ts
export interface InventoryItem {
  id: string;
  identifier: string;
  name: string | null;
  quantity: number | null;
  last_updated: string | null;
}

export interface AddInventoryRequest {
  name: string;
  quantity: number;
}

export interface AddInventoryResponse {
  id: string;
  identifier: string;
}
